File: Lumina-frontend/src/types/book.ts
export interface Book {
    id: string | number;
    title: string;
    author: string;
    genre: string;
    year_published: number;
    description?: string;
    language?: string;
    cover_image_url?: string;
    file_name?: string;
    access_level?: string;
    group_ids?: number[];
    created_at: string;
    updated_at: string;
}

export interface BookCreate {
    title: string;
    author: string;
    genre: string;
    year_published: number;
    description?: string;
    language?: string;
    cover_image_url?: string;
    access_level?: string;
    group_ids?: number[];
}

File: Lumina-backend/luminalib/models/book.py
"""Book model."""

from __future__ import annotations

from sqlalchemy import Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from luminalib.models.base_audit_model import AuditBase


class Book(AuditBase):
    """Represents a library book with ingested content."""

    __tablename__ = "books"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    title: Mapped[str] = mapped_column(String(255), nullable=False, index=True)
    author: Mapped[str] = mapped_column(String(255), nullable=False)
    genre: Mapped[str] = mapped_column(String(100), nullable=False)
    year_published: Mapped[int] = mapped_column(Integer, nullable=False)
    description: Mapped[str | None] = mapped_column(Text)
    file_key: Mapped[str | None] = mapped_column(String(512))
    file_name: Mapped[str | None] = mapped_column(String(255))
    content_type: Mapped[str | None] = mapped_column(String(100))
    file_size: Mapped[int | None] = mapped_column(Integer)
    content_text: Mapped[str | None] = mapped_column(Text)
    summary: Mapped[str | None] = mapped_column(Text)
    review_summary: Mapped[str | None] = mapped_column(Text)
    cover_image_url: Mapped[str | None] = mapped_column(String(1024))
